### ISSUE TITLE
Can ui-router/react support React18 new features?

### ISSUE THREAD

**Issue by Fangfanghao** [ui-router/react#1150]
If we want to support React18 new features ui-router/react, we have to replace
ReactDOM.render
to
createRoot

and replace
unmountComponentAtNode
with
unmount

Do we have a plan to suppport this?

**Comment by kadamgreene** [comment]
Yes, please.  I'm getting errors about missing source maps because of Webpack 5 + React 18.

**Comment by codecdekamal** [comment]
Yes, I am also getting error because of  Webpack 5 + React 18.

**Comment by pahen** [comment]
Any news on this?

**Comment by Akshatjalan** [comment]
Currently No now but, For developers:

Stay informed: Keep an eye on the official announcements from the React and ui-router teams regarding updates and compatibility with React 18.

Check documentation: Review the documentation and release notes of ui-router and React 18 to understand any changes required for compatibility.

Be prepared for updates: Once React 18 is officially released, update your dependencies and check for any potential breaking changes in ui-router or other libraries you are using.

**Comment by andref** [comment]
This issue makes no sense. Ui-router/react does not mount the component tree. You will find no use of `ReactDOM.render` and `unmountComponentAtNode` in its codebase. It works just fine with React 18.

**Comment by chiptus** [comment]
@andref I think they refer to `react-hybrid`: https://github.com/ui-router/react-hybrid/issues/580
